overlord/devicestate/devicestate.go

```diff
@@ -1942,13 +1943,20 @@ func InstallFinish(st *state.State, label string, onVolumes map[string]*gadget.V
 // InstallSetupStorageEncryption creates a change that will setup the
 // storage encryption for the install of the given label and
 // volumes.
-func InstallSetupStorageEncryption(st *state.State, label string, onVolumes map[string]*gadget.Volume) (*state.Change, error) {
+func InstallSetupStorageEncryption(st *state.State, label string, onVolumes map[string]*gadget.Volume, volumesAuth *device.VolumesAuthOptions) (*state.Change, error) {
 	if label == "" {
 		return nil, fmt.Errorf("cannot setup storage encryption with an empty system label")
 	}
 	if onVolumes == nil {
 		return nil, fmt.Errorf("cannot setup storage encryption without volumes data")
 	}
+	if volumesAuth != nil {
+		if err := volumesAuth.Validate(); err != nil {
+			return nil, err
+		}
+		// Auth data must be in memory to avoid leaking credentials.
+		st.Cache(volumesAuthOptionsKey{label}, volumesAuth)
```

Comment:
shouldn't we mark on the tasks that this was set or not? otherwise if there's unexpected restart of snapd how do we detect that we lost information and we need to error/abort the change?

Reply:
Good point, Updated!

I think we have another instance of this in the transition from `install-setup-storage-encryption` to `install-finish` when passing the encryption setup. 

https://github.com/canonical/snapd/blob/f036c4e958c6c06900930842c8b5ee9b54764acd/overlord/devicestate/handlers_install.go#L1299

Reply:
that seems right, can you look into that in a follow-up?

Reply:
Yes, of course